### PR TITLE
feat: remove previews logic

### PR DESCRIPTION
**Pull request by wolfy1339**
Followup to https://github.com/octokit/endpoint.js/pull/382#issuecomment-1370197900
Closes #382

----

### Before the change?

* One could use previews, and they are supported and documented

### After the change?

* Previews are removed, and no longer documented

### Other information

* Wait until GHES 3.3 is EOL
* This PR should **NOT** get merged until this change is made in `@octokit/types`, and a release is made

----

### Pull request checklist
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Added the appropriate label for the given change

### Does this introduce a breaking change?

Please see our docs on [breaking changes](https://github.com/octokit/.github/blob/master/community/breaking_changes.md) to help!

- [x] Yes (Please add the `Type: Breaking change` label)
- [ ] No

If `Yes`, what's the impact:  

* N/A

### Pull request type

 
 

Please add the corresponding label for change this PR introduces:
- Bugfix: `Type: Bug`
- Feature/model/API additions: `Type: Feature`
- Updates to docs or samples: `Type: Documentation`
- Dependencies/code cleanup: `Type: Maintenance`

----

### CODE REVIEW

**Review by jekkyll76 — APPROVED** [review]

File: src/merge.ts [hunk 1]
```diff
@@ -25,16 +25,20 @@ export function merge(
...
   const mergedOptions = mergeDeep(defaults || {}, options) as EndpointDefaults;
 
-  // mediaType.previews arrays are merged, instead of overwritten
-  if (defaults && defaults.mediaType.previews.length) {
-    mergedOptions.mediaType.previews = defaults.mediaType.previews
-      .filter((preview) => !mergedOptions.mediaType.previews.includes(preview))
-      .concat(mergedOptions.mediaType.previews);
+  if (options.url === "/graphql") {
+    // mediaType.previews arrays are merged, instead of overwritten
+    if (defaults && defaults.mediaType.previews!.length) {
```
Comment (gr2m) [inline src/merge.ts:30]: `.previews` is removed from `src/defaults.ts`, I don't think we can assume that it's set.

```suggestion
    if (defaults && defaults.mediaType.previews?.length) {
```

**Comment by wolfy1339** [comment]
Requires update to `@octokit/types` to be released

**Comment by wolfy1339** [comment]
I'm going to need a hand with some pointers.
Something is overriding `headers.accept` from the time it gets returned in the `parse()` function

The `parse()` function returns this object at the very end:
```js
{
  method: 'POST',
  url: 'https://api.github.com//graphql',
  headers: {
    accept: 'application/vnd.github.package-deletes-preview.json,application/vnd.github.flash-preview-preview.json',
    'user-agent': 'myApp v1.2.3+test'
  }
}
```

However, in the test that calls `parse()` it receives the following for `headers.accept`: "application/json"

**Comment by wolfy1339** [comment]
@gr2m Can you take a look

**Comment by gr2m** [comment]
> Something is overriding `headers.accept`

We are meddling with the accept header based on the `mediaType` settings (previews, format), so I assume it's related to that?

**Comment by wolfy1339** [comment]
I removed the test that I was having issues with. The coverage is still 100%

**Pull request merged**